pp.c

```diff
@@ -3717,6 +3717,100 @@ PP_wrapped(pp_substr,
     RETPUSHUNDEF;
 }
 
+/* OP_SUBSTR_LEFT is a specialized version of OP_SUBSTR, where:
+ *     the EXPR is a PADSV
+ *     the OFFSET is a CONST zero
+ *     the replacement pattern is a CONST ""
+ *     it's definitely not in lvalue context (see the check in pp_substr)
+ *     it definitely doesn't have OPpSUBSTR_REPL_FIRST set
+ *     it may be an rvalue or in void context (may support TARGMY later)
+ */
+PP(pp_substr_left)
+{
+    dTARGET;
+    STRLEN curlen;
+    STRLEN utf8_curlen = 0;
+    STRLEN byte_len = 0;
+    SV *sv = PL_stack_sp[-1];
+    const bool rvalue = (GIMME_V != G_VOID) || (PL_op->op_private & OPpTARGET_MY);
+    const bool do_chop = (MAXARG3 == 4);
+    const char *tmps;
+
+    if (SvROK(sv) && do_chop) {
+        Perl_ck_warner(aTHX_ packWARN(WARN_SUBSTR),
+                        "Attempt to use reference as lvalue in substr"
+        );
+    }
+
+    if (do_chop) {
+        SvGETMAGIC(sv);
+        tmps = SvPV_force_nomg(sv, curlen);
+    } else
+        tmps = SvPV_const(sv, curlen);
```

Comment:
You're checking SvROK(sv) before calling SvGETMAGIC() which would be incorrect if sv is tied - I think I would have caught that before, but maybe I missed it.

You could move the reference check after the SvGETMAGIC() inside the do_chop conditional to fix it (and better match pp_substr).